components/availability/src/Availability.svelte

```diff
@@ -171,20 +177,81 @@
     }
   };
 
+  // Consecutive same-availability periods of time, from earliest start_time to latest end_time.
+  const generateEpochs = (
+    slots: SelectableSlot[],
+    participant_threshold: number,
+  ) => {
+    let scale = scaleTime().domain([
+      slots[0].start_time,
+      slots[slots.length - 1].end_time,
+    ]);
+    let epochs = slots
+      .reduce((m, n) => {
+        if (
+          m[m.length - 1] &&
+          JSON.stringify(m[m.length - 1][0].available_calendars) ===
+            JSON.stringify(n.available_calendars)
+        ) {
+          m[m.length - 1].push(n);
+        } else {
+          m.push([n]);
+        }
+        return m;
+      }, [] as TimeSlot[][])
```

Comment:
The main "this is part of the current epoch" vs "this is the start of a new epoch" `.reduce()`.

This does a JSON.stringify() array compare, and I'd be open to alternatives, but in local testing this ran fastest (if cumbersome looking)

Days (+epochs, ticks, timeSlots) generate in < 1ms locally, which feels like a fine benchmark.

Reply:
This comparison will fail if `available_calendars` are ordered differently. I don't think this can happen with the current implementation, but probably worth calling out in a code comment here.

Reply:
Good call; noting in a comment